web/src/overlays.ts

```diff
@@ -50,11 +50,20 @@ function call_hooks(func_list: Hook[]): void {
 }
 
 export function disable_scrolling(): void {
+    // Why disable scrolling?
+    // Since fixed / absolute positined elements don't capture the scroll event unless
+    // they overflow their defined container. Since fixed / absolute elements are treated
+    // as part of the document, we cannot capture `scroll` events on them and prevent propagation
+    // as event bubbling doesn't work naturally.
+    const scrollbar_width = window.innerWidth - document.documentElement.clientWidth;
+    const new_width = `calc(100% - ${scrollbar_width}px)`;
     $("html").css({"overflow-y": "hidden"});
+    $("html, #navbar-fixed-container, #compose").css("width", new_width);
```

Comment:
If we’re doing it this way, we should at least do it in a reasonably modular way. Rather than hard-coding a random list of elements into this obscure file, create a custom property like `:root { --disabled-scrollbar-width: 0px }` that can be used in CSS like `#navbar-fixed-container { width: calc(100% - var(--disabled-scrollbar-width) }`. Then we only need to update the custom property in JS: `$("html").css({"overflow-y": "hidden", "--disabled-scrollbar-width": scrollbar_width})`.

Reply:
Makes sense. Thanks!